src/store/actions.ts

```diff
@@ -68,13 +89,41 @@ export default function createActions(
 				return;
 			}
 
-			let langCode = await langCodeRetriever.getLanguageCodeFromItem( newLanguageItem.id );
+			const langCode = await langCodeRetriever.getLanguageCodeFromItem( newLanguageItem.id );
 
-			if ( typeof langCode === 'string' && !languageCodesProvider.isValid( langCode ) ) {
-				langCode = false;
+			await dispatch( HANDLE_ITEM_LANGUAGE_CODE, langCode );
+		},
+		async [ INIT_FROM_PARAMS ](
+			{ commit, dispatch }: RootContext,
+			params: InitParams,
+		): Promise<void> {
+			if ( params.lemma !== undefined ) {
+				commit( SET_LEMMA, params.lemma );
+			}
+			if ( params.spellVarCode !== undefined ) {
+				commit( SET_SPELLING_VARIANT, params.spellVarCode );
+			}
+			if ( params.language !== undefined ) {
+				commit( SET_LANGUAGE, {
+					id: params.language.id,
+					display: params.language.display,
+					// don’t copy params.language.languageCode here
+				} );
+				await dispatch( HANDLE_ITEM_LANGUAGE_CODE, params.language.languageCode );
```

Comment:
This is not yet doing the right thing for situations where there is no language code in the URL. It currently sets the language code to "undefined" which in this app means that the language code request is still _loading_, but that doesn't actually happen.

So in this case here, if there is no language code, then it should be set to `null`.

You can try this out with the "language item without language code" link in the Readme

Reply:
> So in this case here, if there is no language code, then it should be set to `null`.

Sounds good, let’s do that on the PHP side :P IMHO the current `string | undefined | null | false` type of the `languageCodeFromLanguageItem` is already confusing enough without introducing another type and associated translation logic like “undefined in the init params should become null in the state”.

Reply:
Though we can of course add that to the payload, but it feels redundant.

Reply:
(In that case, I’d change the type declaration here to make the `languageCode` non-optional within the `language`, as long as the `language` itself is present.)

Reply:
> To me "the language code does not exist in the initialization data for the language Item" is true iff "The language code does not exist on the language Item" in the Database. (or is deprecated)

Okay, but that still means we’re missing the `false` case, i.e. the best-rank language code statement has unknown value. I think it would be best if we made the value sent by PHP exactly the same as that returned by `LangCodeRetriever`: it’s present iff a language item was specified at all, either `null` (no / novalue statement), `false` (somevalue statement), or a string (to be validated in JS).

Reply:
There, I think this patch should implement that: https://gerrit.wikimedia.org/r/c/mediawiki/extensions/WikibaseLexeme/+/791622